File: 백준/Bronze/20112. 사토르 마방진/사토르 마방진.py
import sys
input = sys.stdin.readline

n = int(input().rstrip())
maps = [list(input().rstrip()) for _ in range(n)]

maps_reverse = []
for j in range(n):
    now = ""
    for i in range(n):
        now += maps[i][j]
    maps_reverse.append(list(now))

done = 1
for i in range(n):
    if maps[i] != maps_reverse[i]:
        done = 0
        break

print('YES' if done == 1 else 'NO')

File: 백준/Bronze/20112. 사토르 마방진/사토르 마방진.cc
#include <stdio.h>

char str[105][105];
int main(void) {
    int n;
    scanf("%d", &n);
    for (int i = 0; i < n; i++) scanf("%s", str[i]);

    int done = 1;
    for (int idx = 0; idx < n; idx++) {
        for (int c = 0; c < n; c++) {
            if (str[idx][c] != str[c][idx]) {
                done = 0;
                break;
            }
        }
        if (done == 0) break;
    }

    if (done == 1) printf("YES");
    else printf("NO");

    return 0;
}